### DISCUSSION TITLE
Setting ClientID (ContainerID?) when connecting

### DISCUSSION THREAD

**Discussion by joseftw** [Havret/dotnet-activemq-artemis-client#369] [General]
Hello!

Thank you for an awesome lib!

I need to be able to set the ClientID when creating a Consumer, how do I do that?

**Comment by Havret** [comment]
Hi @joseftw,

Thanks for your kind words. :) 

Setting ContainerId on the connection is not possible at the moment, but I can expose it as a part of Connection Factory API.  How about sth like that:

```csharp
var connectionFactory = new ConnectionFactory();
connectionFactory.ContainerIdFactory = () => Guid.NewGuid().ToString();
var endpoint = Endpoint.Create("localhost", 5672, "guest", "guest");
var connection = await connectionFactory.CreateAsync(endpoint);
var consumer = await connection.CreateConsumerAsync(address: "foo", routingType: RoutingType.Multicast);
```

**Reply by joseftw** [reply]
That would be awesome! :)
Either exposing it like that or that you somehow could let us pass a `Open` class to the [ConnectionBuilder](https://github.com/Havret/dotnet-activemq-artemis-client/blob/316afa68917b16aba338e822de64bd210c66bfae/src/ArtemisNetClient/Builders/ConnectionBuilder.cs#L45) (via `ConnectionFactory` somehow)? But maybe that's a bit more complicated?

**Reply by Havret** [reply]
I'd rather not expose AmqpNetLite internals as part of the public API. :)

Please tell me if you're happy with the implementation #371. I changed property name from `ContainerIdFactory` to `ClientIdFactory` as it better fits to the nomenclature that Artemis uses (see #370).

**Reply by joseftw** [reply]
> I'd rather not expose AmqpNetLite internals as part of the public API. :)

I figured ;)

This is awesome, thank you so much!

**Reply by Havret** [reply]
It's released as a part of [2.9.0](https://www.nuget.org/packages/ArtemisNetClient/2.9.0).
